ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java

```diff
@@ -537,295 +315,195 @@ private void loadProperties() {
     }
 
     private void printVersion() {
-        System.out.println("Version: " + this.getVersion());
-        System.out.println("Build Date: " + this.getBuildDate());
+        System.out.println("Version: " + this.version);
+        System.out.println("Build Date: " + this.buildDate);
     }
 
     public void printHelp() {
         // default help menu
         final HelpFormatter helpFormatter = new HelpFormatter();
-        helpFormatter.printHelp("ds3", this.options);
+        helpFormatter.printHelp("ds3_java_cli", this.options);
     }
 
-    public String getBucket() {
-        return this.bucket;
-    }
-
-    void setBucket(final String bucket) {
-        this.bucket = bucket;
-    }
-
-    public String getDirectory() {
-        return this.directory;
-    }
+    // build client from Arguments
+    public Ds3Client createClient() throws MissingOptionException, BadArgumentException {
```

Comment:
With pulling the behavior out of the argument parsing logic this code should be relocated as well.